modules/core/lib/Auth/UserPassBase.php

```diff
@@ -169,6 +169,22 @@ public function isRememberMeChecked() {
 	public function authenticate(&$state) {
 		assert('is_array($state)');
 
+
+		if (isset($state['core:auth:username']) && isset($state['core:auth:password'])) {
+			/* It looks like we are doing HTTP Basic authentication. */
+			$username = $state['core:auth:username'];
+			$password = $state['core:auth:password'];
+
+			if ($this->forcedUsername !== NULL && $this->forcedUsername !== $username) {
+				throw new SimpleSAML_Error_Error('WRONGUSERPASS');
+			}
+
+			$attributes = $this->login($username, $password);
```

Comment:
After testing with a wrong username/password, I've noticed that the ECP-enabled IdP doesn't even reply with an error message. I think the problem is that this exception is not caught, so it escalates up and the code that should handle the error doesn't catch it because the exception is "unknown" to it. But I'm just speculating here...